### PR TITLE
Implement missing batch functions SDK

### LINKED ISSUE #22510: Add updateBatch request to the SDK
### Describe the Bug

The API supports updating multiple items via the ItemsService updateBatch function, but this is not supported in the SDK currently.

### To Reproduce

none

### Directus Version

10.11.0

### Hosting Strategy

Directus Cloud

### Database

_No response_

### PR DESCRIPTION
**Pull request by br41nslug**
Fixes #22510

## Scope

What's changed:

- Added missing updateBatch functions in the SDK

## Potential Risks / Drawbacks

- None im aware of

## Review Notes / Questions

- I would like to lorem ipsum

### CODE REVIEW

**Review by paescuj — COMMENTED** [review]
- [x] Would be great if we could adjust the type for the `items` params for system collections so they reflect the necessity of the primary keys
  ```diff
  + type PartialRequired<T, K extends keyof T> = Partial<T> & { [P in K]: T[P] };
  
  /**
   * Update multiple dashboards as batch.
   * @param items
   * @param query
   * @returns Returns the dashboard objects for the updated dashboards.
   * @throws Will throw if no items are provided
   */
  export const updateDashboardsBatch =
  	<Schema, const TQuery extends Query<Schema, DirectusDashboard<Schema>>>(
  -	 	items: Partial<DirectusDashboard<Schema>>[],
  +	 	items: PartialRequired<DirectusDashboard<Schema>, 'id'>[],
  ```
  
- [x] Not sure if we really need/want to enforce the `throwIfEmpty` checks as the API actually accepts empty arrays as input (and returns `[]`)
- [x] Would be great to add a quick example to https://docs.directus.io/guides/sdk/getting-started.html

**Comment by br41nslug** [comment]
> * [ ]  Would be great if we could adjust the type for the `items` params for system collections so they reflect the necessity of the primary keys

Interesting improvement suggestion :d Please make an issue for that, happy to address that in a separate pr

>     * [ ]  Not sure if we really need/want to enforce the `throwIfEmpty` checks as the API actually accepts empty arrays as input (and returns `[]`)

But what would be the point of doing a batch update call with nothing to update? 🤔 I added the throw more as a indication for the end users that something might be wrong with the call they are doing. But no they are not technically required.

>     * [ ]  Would be great to add a quick example to https://docs.directus.io/guides/sdk/getting-started.html

There is a separate issue for that to add more complete documentation for these batch functions beyond a quick example https://github.com/directus/directus/issues/22516

**Comment by paescuj** [comment]
> > * [x]   Would be great if we could adjust the type for the `items` params for system collections so they reflect the necessity of the primary keys
> 
> Interesting improvement suggestion :d Please make an issue for that, happy to address that in a separate pr

Yeah, happy to create an issue for that if we're going to expand and enable definition of primary keys on custom collections as well (e.g. via virtual `primaryKey` type, similar to `csv`/`json`/`datetime`). Otherwise I'd prefer to implement it as part of this PR since we're now in the right context and it would be a small change (as in the given example).

> 
> > * [ ]   Not sure if we really need/want to enforce the `throwIfEmpty` checks as the API actually accepts empty arrays as input (and returns `[]`)
> 
> But what would be the point of doing a batch update call with nothing to update? 🤔 I added the throw more as a indication for the end users that something might be wrong with the call they are doing. But no they are not technically required.

Probably nothing, just wondering if we should add restrictions in the SDK that doesn't exist in the API?

> > * [x]   Would be great to add a quick example to https://docs.directus.io/guides/sdk/getting-started.html
> 
> There is a separate issue for that to add more complete documentation for these batch functions beyond a quick example #22516

Okay cool, updated the issue to include all batch methods 👍

**Comment by br41nslug** [comment]
> Yeah, happy to create an issue for that if we're going to expand and enable definition of primary keys on custom collections as well (e.g. via virtual primaryKey type, similar to csv/json/datetime). Otherwise I'd prefer to implement it as part of this PR since we're now in the right context and it would be a small change (as in the given example).

I qualify this as a new feature not currently present in the SDK, with indeed more than a tiny tweak of lift required to apply it consistently across the SDK, so a new issue or FR is the way to go here.

**Comment by br41nslug** [comment]
> I think that's only an issue for those new update batch functions, for the existing update functions the primary key is part of the path, thus already required.

I thought there were some of these same situations in creating items for system collections (creating collections springs to mind)

**Comment by paescuj** [comment]
> > I think that's only an issue for those new update batch functions, for the existing update functions the primary key is part of the path, thus already required.
> 
> I thought there were some of these same situations in creating items for system collections (creating collections springs to mind)

Okay, will create an issue so we can check for other places.

File: sdk/src/rest/commands/update/dashboards.ts [hunk 1]
```diff
@@ -34,6 +34,29 @@ export const updateDashboards =
...
+/**
+ * Update multiple dashboards as batch.
+ * @param items
+ * @param query
+ * @returns Returns the dashboard objects for the updated dashboards.
+ * @throws Will throw if no items are provided
+ */
+export const updateDashboardsBatch =
+	<Schema, const TQuery extends Query<Schema, DirectusDashboard<Schema>>>(
+		items: Partial<DirectusDashboard<Schema>>[],
```
Comment (paescuj) [inline sdk/src/rest/commands/update/dashboards.ts:46]: I think we need to use `NestedPartial` for most of the update batch functions, as nested objects can be, for example
- an update (only containing primary key + fields to update)
  ```js
  updateDashboardsBatch([
    {
      id: "0fed4f30-b500-4457-9c71-e00797ce973b",
      user_created: {
        id: "bb9342a4-9454-4cb5-8914-c80d720e4e02",
        first_name: "updated via update batch",
      },
    },
  ])
  ```
- a new item creation (not containing primary key + probably omitting other fields)
  ```js
  updateDashboardsBatch([
    {
      id: "0fed4f30-b500-4457-9c71-e00797ce973b",
      user_created: {
        first_name: "created via update batch",
      },
    },
  ])
  ```

```suggestion
		items: NestedPartial<DirectusDashboard<Schema>>[],
```

**Review by paescuj — APPROVED** [review]

**Pull request merged**